src/arrayMethodJoin.js

```diff
@@ -5,7 +5,23 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (i > 0 && separator === undefined) {
+        result += ',';
+      } else if (i > 0 && separator !== undefined) {
+        result += separator;
+      }
```

Comment:
You can set a default value for your `separator` argument (`function(separator = ',')`). After this, you can check if the current iteration is not last and add this separator after `if` block below.